### PR TITLE
improve UI

### PR DESCRIPTION
**Pull request by flepied**
- link to github PR/gerrit changes and author

### CODE REVIEW

**Comment by flepied** [comment]
@morucci I don't know how to do it on the "Events Stats" tab

File: web/src/components/common.js [hunk 1]
```diff
@@ -5,6 +5,27 @@ import Col from 'react-bootstrap/Col'
 import Card from 'react-bootstrap/Card'
 
 
+function change_url(x, name=null) {
```
Comment (morucci) [inline web/src/components/common.js]: We should probably add the review system base url in each Change and Change events in the DB. And makes the web UI able to react on that. We should be able to handle any gerrit instances as well and also any github enterprise deployments.

File: web/src/components/top.js [hunk 2]
```diff
@@ -34,7 +34,7 @@ class TopEventsTable extends React.Component {
                 <tbody>
                   {this.props.data.map((x, index) =>
                     <tr key={index}>
-                      <td>{index}</td>
+                      <td>{index + 1}</td>
```
Comment (morucci) [inline web/src/components/top.js]: This change could be done in another PR imo.

File: web/src/components/common.js (hunk 1 above)
Comment (flepied) [reply web/src/components/common.js]: sure it would be cleaner

File: web/src/components/top.js (hunk 2 above)
Comment (flepied) [reply web/src/components/top.js]: ok

File: web/src/components/common.js (hunk 1 above)
Comment (flepied) [reply web/src/components/common.js]: @morucci where should I do that in the code?

File: web/src/components/top.js (hunk 2 above)
Comment (flepied) [reply web/src/components/top.js]: done

File: web/src/components/common.js (hunk 1 above)
Comment (flepied) [reply web/src/components/common.js]: I tried to do it for github pr. let me know if it was the right way

File: monocle/github/pullrequest.py [hunk 3]
```diff
@@ -257,6 +258,10 @@ def extract_pr_objects(pr):
                 change['repository_fullname'],
                 change['number'],
             )
+            change['url'] = ('https://%s/%s/pull/%s'
```
Comment (morucci) [inline monocle/github/pullrequest.py]: I would have instead added three fields:
- host: could be github.com or github.company.com or gerrit.ovirt.com or softwarefactory-project.io
- host_type: could be gerrit or github
- host_base_url: should be None usually for most services accessible from '/' but must be '/r' for softwarefactory-project.io for instance.

From those fields, in the UI, we can compute the PR web url.

This three fields must be added in db.py self.index. A index deletion + re-index will be needed. And host_base_url it should be an option CLI argument in the crawler UI in bot gerrit and github crawler I guess.

Let me know if you think about a better way to achieve this.

Comment (flepied) [reply monocle/github/pullrequest.py]: I think I'll prefer to have the url computed on the backend side to have the simplest API. Let's discuss when you'll have time.

Comment (morucci) [reply monocle/github/pullrequest.py]: Ok fine to me, the change url could be computed in the in the crawler and be part of object in the DB. What about the url to access user profils ? Should it be computed in the same way ? I'd say yes if we move this way. We should also take care of the host_base_url.

Comment (flepied) [reply monocle/github/pullrequest.py]: user profiles should be done the same way I'd say too. Should we just pass a base_url  to the crawlers?

Longer term we could also have internal pages for users and changes.

Comment (morucci) [reply monocle/github/pullrequest.py]: Yes. And looking more at the code. I see we could use the existing --host option. For the gerrit crawler it is well handled, I can call the gerrit crawler this way:
```
python3 ./monocle/main.py gerrit_crawler --repository "^software-factory/.*" --updated-since "2019-01-01 00:00:00" --loop-delay 60 --host https://softwarefactory-project.io/r
```

But for github, it is not consistent because it does not include the scheme:
```
monocle github_crawler --host github.com --token XXX --org docker --updated-since 2020-03-01
```
So it should be a complete url with the scheme, and looking at graphql.py self.url is hardcoded. So this should be fixed as well.

So yes we could use --host directly.

File: monocle/gerrit/review.py [hunk 4]
```diff
@@ -40,8 +40,8 @@ class ReviewesFetcher(object):
 
     log = logging.getLogger("monocle.ReviewesFetcher")
 
-    def __init__(self, host, repository_prefix):
-        self.host = host
+    def __init__(self, base_url, repository_prefix):
+        self.base_url = base_url
```
Comment (morucci) [inline monocle/gerrit/review.py]: This could be renamed to base_url in the CLI as well.

File: monocle/github/pullrequest.py [hunk 5]
```diff
@@ -257,6 +258,10 @@ def extract_pr_objects(pr):
                 change['repository_fullname'],
                 change['number'],
             )
+            change['url'] = ('%s/%s/pull/%s'
```
Comment (morucci) [inline monocle/github/pullrequest.py]: You need to add "url" in elastic schema in db.py.

**Comment by morucci** [comment]
Also could it be splitted in two PRs ?:
* backend change
* ui change

File: monocle/gerrit/review.py [hunk 6]
```diff
@@ -130,6 +130,7 @@ def extract_pr_objects(review):
                 'repository_fullname': review['project'],
                 'repository_shortname': "/".join(
                     review['project'].split('/')[1:]),
+                'url': '%s/c/gerrit/+/%s' % (self.base_url, review['_number']),
```
Comment (morucci) [inline monocle/gerrit/review.py]: This works better:
'url': '%s/%s' % (self.base_url, review['_number']),

File: monocle/github/pullrequest.py (hunk 5 above)
Comment (morucci) [reply monocle/github/pullrequest.py]: See this change: https://github.com/morucci/monocle/pull/28

File: monocle/gerrit/review.py (hunk 6 above)
Comment (flepied) [reply monocle/gerrit/review.py]: done

File: monocle/github/pullrequest.py (hunk 5 above)
Comment (flepied) [reply monocle/github/pullrequest.py]: ok

File: monocle/gerrit/review.py (hunk 4 above)
Comment (flepied) [reply monocle/gerrit/review.py]: done

File: monocle/github/pullrequest.py (hunk 3 above)
Comment (flepied) [reply monocle/github/pullrequest.py]: is it ok now?

File: web/src/components/common.js [hunk 7]
```diff
@@ -5,6 +5,25 @@ import Col from 'react-bootstrap/Col'
...
+  if (!name) {
```
Comment (morucci) [inline web/src/components/common.js]: Could you move the JS part in another PR ?

Comment (flepied) [reply web/src/components/common.js]: I split it in 2 commits

**Comment by morucci** [comment]
I've tested it. It works with a quick change in the common.js. Expect the author_url, that might require a bit more work. So I merge the PR. I'll followup with a quick fix.

**Comment by morucci** [comment]
See #31
